buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -466,48 +472,67 @@ private void logProcessInfo(String prefix, ProcessHandle.Info info) {
     }
 
     private void logFileContents(String description, Path from) {
-        logger.error("{} `{}`", description, this);
-        long lineCutoff;
-        try(Stream<String> lines = Files.lines(from, StandardCharsets.UTF_8)) {
-            lineCutoff = lines.count();
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-        final Map<String, Long> errorsAndWarnings;
-        try(Stream<String> lines = Files.lines(from, StandardCharsets.UTF_8)) {
-            errorsAndWarnings = lines
-                .filter(line -> line.contains("ERROR") || line.contains("WARN"))
-                .map(this::normalizeLogLine)
-                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-        lineCutoff -= 50;
+        final Map<String, Integer> errorsAndWarnings = new LinkedHashMap<>();
+        LinkedList<String> ring = new LinkedList<>();
```

Comment:
I'm not sure I understand the name "ring" here.

Reply:
It's a [circular buffer](https://en.wikipedia.org/wiki/Circular_buffer), although we are using it more like a deque without its second level indirection overhead.